pheps/phep-0001.md

```diff
@@ -0,0 +1,287 @@
+
+Standards Track PHEPs may consist of two parts, a design document and a reference implementation. It is generally recommended that at least a prototype implementation in an existing or new PyHC package be co-developed with the PHEP, as ideas that sound good in principle sometimes turn out to be impractical when subjected to the test of implementation.
+
+## Discussing a PHEP
+As soon as the draft PHEP is submitted in an open PR and a PHEP number has been assigned, a discussion thread for the PHEP must be created to provide a central place to discuss and review its contents, and the PHEP must be updated so that the `Discussions-To` header links to it. The venue for this discussion must be the GitHub discussion thread for the PR with the draft PHEP. The discussion must also be announced by the PHEP authors via reasonable venues; this includes (but is not necessarily limited to) the PyHC email list and the PyHC Element/Slack.
+
+Informal conversation may happen in the PyHC mailing list, Element/Slack, or private side channels. However, the discussion of record is the public thread on the pull request. Substantive input into the PHEP or questions about it belong in the PR comments and any side conversations should ultimately result in comments on the PR. This ensures everyone can follow and contribute, avoids fragmenting the discussion, and makes sure it is fully considered as part of the PHEP review process. Comments, support, concerns and other feedback on this designated thread are a critical part of what the community will consider when reviewing the PHEP. Individual comments or the GitHub "review" process may both be used. Commenters may use the GitHub "approve" feedback to indicate they believe the PHEP is ready for a vote; however, this does not change the PHEP status to "Accepted".
+
+PHEP authors should incorporate community feedback by making edits and additional pushes to their branch. They must add a new `Post-History` entry added with each push. It is recommended to address as much actionable feedback as practicable with each push; however, each push may certainly contain multiple commits for ease of review. PHEP authors must not rewrite history after a push, i.e. no force-pushing. Rebasing of history before a push for greater clarity is acceptable.
+
+PHEP authors are also responsible for managing the progress of the discussion of their PHEP, for instance by updating the PR description to highlight topics of particular interest and to provide the dates of votes on the PHEP resolution.
+
+If a PHEP undergoes a significant re-write or other major, substantive changes to its proposed specification, the author(s) and editor may discuss closing the existing PR and opening a new one, starting from the changed version. If this occurs, the `Discussions-To` link must be updated.
```

Comment:
I'm not tied to this...either we should require it to always stay in the same PR, or we should set some rules around that (e.g. are the initial commits squashed?) But I can see the conversation getting so tangled it's worth applying the comments and starting over.

Reply:
This is probably situation dependent, so my inclination would be to allow for flexibility in this.

Reply:
+1 to flexibility. I'd wanna see the process in action a few times before setting hard rules. Although my gut reaction is to want to keep things in the same PR (it should be simple enough to see the original PR state via commits when desired).

Reply:
Sounds like the wording should be left as-is? Ultimately it's up to authors and editor to make a decision; the important thing is the discussions-to link.

Reply:
Yeah I'd say we can leave it as-is.